### DISCUSSION TITLE
Netbox UI is throwing Module Not found Error

### DISCUSSION THREAD

**Discussion by neharaina1771** [netbox-community/netbox#14990] [Help Wanted!] [answered]
Hi Team ,

Urgent help required.

I recently added the self signed certificate to run the netbox on https .Netbox is working in docker container on https,
but when I deployed the container in minikube then it is throwing below error .
Please help me to understand how to debug the issue as there exist no module with name 'no'.

Server Error
A module import error occurred during this request. Common causes include the following:

Missing required packages. This installation of NetBox might be missing one or more required Python packages. These packages are listed in requirements.txt and local_requirements.txt, and are normally installed as part of the installation or upgrade process. To verify installed packages, run pip freeze from the console and compare the output to the list of required packages.

WSGI service not restarted after upgrade. If this installation has recently been upgraded, check that the WSGI service (e.g. gunicorn or uWSGI) has been restarted. This ensures that the new code is running.

The complete exception is provided below:

<class 'ModuleNotFoundError'>

No module named 'no'

Python version: 3.10.12
NetBox version: 3.6.5
Plugins: None installed

**Comment by llamafilm** [comment]
It sounds like you have a typo somewhere, search your code for the word `no ` (with a space after it).
You can also set `DEBUG=True` in configuration.py and that should provide more detail on the source of the issue.

**Reply by neharaina1771** [reply]
Hi ,

I searched in code ..but nothing found with 'no' .. 

DEBUG = _environ_get_and_map('DEBUG', 'False', _AS_BOOL) this is set in conf.py file

**Reply by neharaina1771** [reply]
The same code is working in docker container bui when i push the container to minikube then this issue is occuring .If this is the code issue then it would not have working in docker container also.

**Reply by neharaina1771** [reply]
From inside the pod , when I curl the url , server is giving 500 error from inside pod

I have no name!@netbox-657976cc47-jbvdx:/opt/netbox/netbox$ ./manage.py runserver
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'
Performing system checks...

System check identified no issues (0 silenced).
January 31, 2024 - 04:27:00
Django version 4.2.7, using settings 'netbox.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.

[31/Jan/2024 04:27:40] "GET / HTTP/1.1" 500 2293

[31/Jan/2024 04:28:45] "GET / HTTP/1.1" 500 2293
[31/Jan/2024 04:29:37] "GET / HTTP/1.1" 500 2293

**Comment by neharaina1771** [comment]
<img width="1637" alt="Screenshot 2024-01-31 at 9 50 02 AM" src="https://github.com/netbox-community/netbox/assets/47520666/ec27b40e-5907-426c-8e78-dd662fd2d73a">

**Comment by candlerb** [comment]
If you're running in minikube, then where did you get the YAML resource definitions from? Can you show them? I suspect there is a bug in them.

[Beware YAML](https://www.infoworld.com/article/3669238/7-yaml-gotchas-to-avoidand-how-to-avoid-them.html) with values like true, false, yes, no. If you don't quote them then they may be treated as booleans (*). Since your error message involves the string "no" then I suspect this might be what's going on.

You can set environment variable DEBUG to "true" using standard kubernetes mechanisms. This should show you the full backtrace within the web UI, which will identify the exact line of python code which is problematic. Again, make sure the value is quoted in YAML.

```
        ...
        spec:
          env:
            - name: DEBUG
              value: "true"
          image: netbox:blah
          ...
```

(*) This was changed in YAML 1.2 to be only "true" or "false". However there are still many YAML 1.1 parsers out there.

**Reply by neharaina1771** [reply]
I am using the helm template to deploy the image to minikube

**Reply by neharaina1771** [reply]
I am checking in template if there is anything with 'no'

**Reply by neharaina1771** [reply]
that issue got resolved .Thanks a lot for your help .Now encountering another issue 

<class 'AttributeError'>

'NoneType' object has no attribute 'rsplit'

Python version: 3.10.12
NetBox version: 3.6.5
Plugins: None installed

**Reply by candlerb** [reply, marked as answer]
> I am using the helm template to deploy the image to minikube

"The helm template" - which helm template?? This must be some third-party creation, and a quick Google turns up many of them, so you should be specific about which one.

In any case, the problems you have deploying Netbox using it would be best raised with the creator of that helm template, not with the Netbox project itself.

> Now encountering another issue

As before: you could set DEBUG=true and get a backtrace for this error. 

But again, since you've already found that Netbox deploys successfully using docker-compose, your problem is still with the Helm chart or how you're using it - not with Netbox itself or the published docker image.

**Reply by neharaina1771** [reply]
Thanks for the help and support..
